src/main/java/chess/domain/board/Step.java

```diff
@@ -33,11 +29,19 @@ public boolean hasEnemy() {
         return locationState.isEnemy();
     }
 
+    public boolean hasAlly() {
+        return locationState.isAlly();
```

Comment:
이런 부분들은 사실 다음과 같이 작성해도 될 것 같은데요.
```suggestion
        return locationState == LocationState.ALLY;
```

최대한 객체에 메시지를 보내도록 구현하기 위해 조금 과하게 메서드를 만든 것 같기도 합니다.
두 개의 차이가 있을까요?? 이런 상황에도 객체에 메시지를 보내도록 해야 할까요?

Reply:
== 비교로도 충분할 것 같아요!
그냥 취향 차이 정도 일것 같네요